src/VirtoCommerce.SitemapsModule.Data/BackgroundJobs/SitemapProcessJob.cs

```diff
@@ -0,0 +1,235 @@
+            using (var stream = SystemFile.Open(localTmpPath, FileMode.CreateNew))
+            using (var zipArchive = new ZipArchive(stream, ZipArchiveMode.Create, true))
+            {
+                // Create default sitemap.xml
+                await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, ModuleConstants.SitemapFileName, SendNotificationWithProgressInfo);
+
+                var sitemapUrls = await _sitemapXmlGenerator.GetSitemapUrlsAsync(storeId, baseUrl);
+                foreach (var sitemapUrl in sitemapUrls.Where(url => !string.IsNullOrEmpty(url)))
+                {
+                    await CreateSitemapPartAsync(zipArchive, storeId, baseUrl, sitemapUrl, SendNotificationWithProgressInfo);
+                }
+            }
+
+            //Copy export data to blob provider for get public download url
+            using (var localStream = SystemFile.Open(localTmpPath, FileMode.Open, FileAccess.Read))
+            using (var blobStream = _blobStorageProvider.OpenWrite(relativeUrl))
+            {
+                await localStream.CopyToAsync(blobStream);
+            }
+
+            // Add unique key for every link to prevent browser caching
+            notification.DownloadUrl = $"{_blobUrlResolver.GetAbsoluteUrl(relativeUrl)}?v={DateTime.UtcNow.Ticks}";
+            notification.Description = "Sitemap download finished";
+        }
+        catch (Exception exception)
+        {
+            notification.Description = "Sitemap download failed";
+            notification.Errors.Add(exception.ExpandExceptionMessage());
+        }
+        finally
+        {
+            notification.Finished = DateTime.UtcNow;
+            await _notifier.SendAsync(notification);
+        }
+    }
+
+    private async Task ProcessStore(Store store)
+    {
+        var outputAssetFolder = string.Format(Core.ModuleConstants.StoreAssetsOutputFolderTemplate, store.Id);
+
+        _logger.LogInformation("Starting export {sitemapUrl} for store {storeId} to {outputAssetFolder}.", Core.ModuleConstants.SitemapFileName, store.Id, outputAssetFolder); // Log success
+
+        await ExportSitemapPartAsync(store.Id, store.Url, outputAssetFolder, Core.ModuleConstants.SitemapFileName, null);
+
+        foreach (var sitemapUrl in await _sitemapXmlGenerator.GetSitemapUrlsAsync(store.Id, store.Url))
+        {
+            _logger.LogInformation("Starting export {sitemapUrl} for store {storeId} to {outputAssetFolder}.", sitemapUrl, store.Id, outputAssetFolder); // Log success
+
+            await ExportSitemapPartAsync(store.Id, store.Url, outputAssetFolder, sitemapUrl, null);
+        }
+    }
+
+    private async Task CreateSitemapPartAsync(ZipArchive zipArchive, string storeId, string baseUrl, string sitemapUrl, Action<ExportImportProgressInfo> progressCallback)
+    {
+        var sitemapPart = zipArchive.CreateEntry(sitemapUrl, CompressionLevel.Optimal);
+        using var sitemapPartStream = sitemapPart.Open();
```

Comment:
I would add an empty line between 219 and 220 lines . The code looks similar to the same in the ExportSitemapPartAsync method, but stylistically designed differently. It attracts attention.

Reply:
Fixed